packages/transit-vehicle-overlay/src/VehicleMarker/index.js

```diff
@@ -0,0 +1,118 @@
+import React from "react";
+import ReactDOMServer from "react-dom/server";
+import PropTypes from "prop-types";
+
+import L, { divIcon } from "leaflet";
+import { withLeaflet } from "react-leaflet";
+import RotatedMarker from "./RotatedMarker";
+
+import VehiclePopup from "./popup";
+import VehicleToolTip from "./tooltip";
+import { vehicleType } from "../types";
+import * as utils from "../utils";
+import * as Styled from "./styled";
+
+/**
+ * This component demonstrates a custom marker used in the Vehicles overlay provided as
+ * an example. It is modeled after the component available in this file:
+ * https://github.com/OpenTransitTools/transit-components/blob/master/lib/vehicles/VehicleMarker.js
+ */
+function VehicleMarker(props) {
+  const { vehicle } = props;
+  const { hasTooltip } = props;
+  const { hasPopup } = props;
+  const { tracked } = props;
+  const { setTracked } = props;
+  const { color } = props;
+
+  const { leaflet } = props;
+
+  const { closeZoom, midZoom } = props;
+  const { midSize, farSize } = props;
+
+  const position = [vehicle.lat, vehicle.lon];
+  const zPos = tracked ? 1000 : 0;
+  const heading = utils.checkHeading(vehicle.heading);
+
+  function makeIcon() {
+    const zoom = utils.getZoom(leaflet);
+
+    let icon = null;
+    if (zoom >= closeZoom) {
+      icon = Styled.makeVehicleIcon(vehicle.routeType, color, tracked);
+    } else {
+      const size = zoom >= midZoom ? midSize : farSize;
+      const iconHtml = ReactDOMServer.renderToStaticMarkup(
+        tracked ? (
+          <Styled.TrackedVehicleCircle size={size} colorselected={color} />
+        ) : (
+          <Styled.VehicleCircle size={size} colorselected={color} />
+        )
+      );
+      icon = divIcon({ className: "", html: iconHtml });
+    }
+
+    return icon;
+  }
+
+  const icon = makeIcon();
```

Comment:
It'd probably be a good idea to cache these icons, but that's probably something for a later effort.